app/javascript/components/Main/Main.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import Header from '../Header/Header'
+import FiltersBar from '../FiltersBar/FiltersBar'
+import RestaurantsMain from '../Restaurants/RestaurantsMain/RestaurantsMain'
+
+export default class Main extends Component {
+  state = { restaurants: [], cuisins: [], filters: { rating: null, cuisine: null, speed: null } };
+
+  componentDidMount() {
+    this.fetchRestaurants();
+    this.fetchCuisins();
+  }
+
+  fetchRestaurants() {
+    fetch('/restaurants')
+      .then(
+        response => response.json(),
+        error => console.log('An error occurred fetching restaurants.', error)
+      )
+      .then(restaurants => this.setState({ restaurants }))
+  }
+
+  fetchCuisins() {
+    fetch('/genres')
+      .then(
+        response => response.json(),
+        error => console.log('An error occurred fetching genres.', error)
+      )
+      .then(cuisins => this.setState({ cuisins }))
+  }
+
+  filterCuisins(e){
+    const cuisine = e.target.value
+
+    this.setState({filters: {...this.state.filters, cuisine}});
```

Comment:
It's better to use functional `setState`, especially in this cases: https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b